### ISSUE TITLE
Introduce AccountSet class

### ISSUE THREAD

**Issue by kennell** [kennell/rai#3]
Currently, the `accounts` property of a `Wallet` instance simply returns a list of `Account` instances.

I am thinking about implementing a holder class named `AccountSet` which contains a set of `Account`'s and returning that as the Wallet `accounts` property. This would allow for a cleaner API in some cases, e.g. with such a class one could do things like:

```python
wallet = Wallet(id='AAAABBBBCCCCDDDD')
new_acc = wallet.accounts.new()
acc = wallet.accounts['xrb_123123123123123']
```

And by implementing `__iter__` or a generator  it would still be possible to iterate over all accounts in a comfortable way as it is right now:

```python
wallet = Wallet(id='AAAABBBBCCCCDDDD')
for a in wallet.accounts:
    print(a.balance)
```

Any input/ideas on this is appreciated @crolfe @jordancoil @jdistler

**Comment by crolfe** [comment]
You could keep `accounts` as a property on wallet, and add a `create_account` method to wallet.  The wallet id is needed for `account_create` command, so an `AccountSet` class would still need to know about the `Wallet` class.

One enhancement I think should be considered either way is caching the accounts (e.g in an `_accounts` attribute), as they shouldn't change all that often (depending on use case).  Right now, anytime you access this property, you will be making a network call, which could be costly if your node isn't on the same machine.  This behaviour could also be controlled by a setting, if necessary.

E.g. something like: 
```    
    @property                                                              
    def accounts(self):                                                         
        """                                                                     
            Returns all accounts in the wallet                                  
        """                                
        if self._accounts is None:
            self._accounts = set(self._get_accounts())
        return self._accounts
```

**Comment by kennell** [comment]
Yes, that is the alternative, having methods like `wallet.create_account()`. Yes caching is also something i have been thinking about, looking to do that once it's nearly feature complete. 

So for now i will keep the current approach with the list, keeping this open for discussion.
